packages/core/src/pubsub/outstanding-queries.ts

```
@@ -0,0 +1,129 @@
+    private _minutesThreshold = 1; 
+
+    add(id: string, query: Query) : Boolean {
+      //enforce no duplicate outstanding queries
+      if(this.queryMap.get(id) != undefined){ throw new Error("Error: Duplicate Outstanding Query Cannot be added") }
+      try{
+        // add to map
+        this.queryMap.set(id, query);
+        // add to queue
+        this.queryQueue.enqueue(query);
+        return true;
+      }catch(e){
+        const errorMessage = `Error in OutstandingQueries.add(), ${e.message}`
+        console.error(errorMessage)
+        throw new Error(errorMessage)
+      }
+    }
+
+    remove(topQuery: Query) : Boolean {
+      try{
+        //remove queryId key for top query
+        this.queryMap.delete(topQuery.queryID);
+        //dequeue top query
+        this.queryQueue.dequeue();
+        return true;
+      }catch(e){
+        const errorMessage = `Error in OutstandingQueries.remove(), ${e.message}`
+        console.error(errorMessage)
+        throw new Error(errorMessage)
+      }
+    }
+
+    /**
+     * Event-driven method to clean up outdates outstanding queries
+     * @param
+     * @public
+     */
+    cleanUpExpiredQueries(){
+
+        if ( this.queryQueue.size() > 0 ) {
+
+          const topQuery: Query = this.queryQueue.front()
+
+          //var diffMs = (topQuery.timestamp - Date.now()); // milliseconds 
+          const diffMs = (Date.now() - topQuery.timestamp); // milliseconds
+          
+          const differenceInMinutes = Math.round(((diffMs % 86400000) % 3600000) / 60000); // minutes
+          
+          if(differenceInMinutes > this._minutesThreshold){
+              try{
+                this.remove(topQuery);
+                //recursively check if the most prioritized query can be cleaned
+                this.cleanUpExpiredQueries()
+              }catch(e){
+                const errorMessage = `Error in OutstandingQueries.cleanUpExpiredQueries(), ${e.message}`
```

Smells not good. 1) I fail to see what benefit it provides while hiding an original error. 2) Please, use logger. Or do not log here, instead relying on error messages in a hope that a caller would log the error.